pkg/resolver/search.go

```diff
@@ -526,39 +548,44 @@ func WhereClauseFilter(input *model.SearchInput) []exp.Expression {
 		}
 	}
 	if input.Filters != nil {
+
 		for _, filter := range input.Filters {
 			if len(filter.Values) > 0 {
 				values := pointerToStringArray(filter.Values)
 
-				// If property is of array type like label, remove the equal sign in it and use colon
-				// - to be similar to how it is stored in the database
-				if _, ok := arrayProperties[filter.Property]; ok {
-					cleanedVal := make([]string, len(values))
-					for i, val := range values {
-						labels := strings.Split(val, "=")
-						if len(labels) == 2 {
-							cleanedVal[i] = fmt.Sprintf(`{"%s":"%s"}`, labels[0], labels[1])
-						} else if len(labels) == 1 {
-							//// If property is of array type, format it as an array for easy searching
-							cleanedVal[i] = labels[0]
-						} else {
-							klog.Error("Error while decoding label string")
-							cleanedVal[i] = val
-						}
+				for key, val := range propTypeMap {
+					if key == filter.Property { //check if property exists in dataTypeMap to get datatype
+						klog.V(5).Infof("Prop in map:%s, filter prop is: %s, datatype :%s\n", key, filter.Property, val)
+						dataType = val
+
+						cleanedVal := make([]string, len(values))
 
+						for i, val := range values {
+							if dataType == "object" { //map[string]interface{}
+								labels := strings.Split(val, "=")
+								cleanedVal[i] = fmt.Sprintf(`{"%s":"%s"}`, labels[0], labels[1])
+							} else if dataType == "array" { //[]string
+								cleanedVal[i] = fmt.Sprintf(`["%s"]`, val)
+
+							} else {
+								klog.Error("Error while decoding label string")
```

Comment:
Add more information to help debug.
```suggestion
								klog.Errorf("Error while decoding label string with dataType %s", dataType)
```